### ISSUE TITLE
Workaround for compiling firefox with LTO

### ISSUE THREAD

**Issue by 12101111** [gentoo/musl#297]
I have managed to compile firefox with lto.
![screenshot-2020-02-21-08-56-59](https://user-images.githubusercontent.com/8438475/75021938-240a5400-548d-11ea-9e60-092412bd6a3c.png)

The keypoint is compile `clang` and `lld` with more stack size.
As [documents from musl](https://wiki.musl-libc.org/functional-differences-from-glibc.html#Thread-stack-size) say, The default stack size on musl is 80KB, while glibc gives a much more big stack (2MB -10MB).
So stack overflow happend when link `libxul.so`
The default stack size of a program can be set on link time use `-Wl,-z,stack-size=N` or use binary patch tool: https://github.com/yaegashi/muslstack.
Upstream report: https://github.com/llvm/llvm-project/issues/75

**Comment by 12101111** [comment]
![图片](https://user-images.githubusercontent.com/8438475/75105817-3504c200-5653-11ea-95aa-bc4a3131a908.png)
PGO also works, but have some issues about sandbox:
```log
F: open_wr
S: deny
P: /dev/dri/renderD128
A: /dev/dri/renderD128
R: /dev/dri/renderD128
C: /tmp/portage/www-client/firefox-73.0/work/firefox-73.0/ff/instrumented/dist/firefox/firefox
data:text/html,<script>Quitter.quit()</script> -profile /tmp/portage/www-client/firefox-73.0/te
mp/tmpS0xOPm

F: open_wr
S: deny
P: /dev/dri/card0
A: /dev/dri/card0
R: /dev/dri/card0
C: /tmp/portage/www-client/firefox-73.0/work/firefox-73.0/ff/instrumented/dist/firefox/firefox
data:text/html,<script>Quitter.quit()</script> -profile /tmp/portage/www-client/firefox-73.0/te
mp/tmpS0xOPm

F: open_wr
S: deny
P: /dev/dri/renderD128
A: /dev/dri/renderD128
R: /dev/dri/renderD128
C: /tmp/portage/www-client/firefox-73.0/work/firefox-73.0/ff/instrumented/dist/firefox/firefox http://localhost:8888/index.html -profile /tmp/portage/www-client/firefox-73.0/temp/tmpS0xOPm

F: open_wr
S: deny
P: /dev/dri/card0
A: /dev/dri/card0
R: /dev/dri/card0
C: /tmp/portage/www-client/firefox-73.0/work/firefox-73.0/ff/instrumented/dist/firefox/firefox http://localhost:8888/index.html -profile /tmp/portage/www-client/firefox-73.0/temp/tmpS0xOPm
```
And some changes in ebuild to get it work:
```
@@ -263,6 +275,13 @@
 	# Allow user to apply any additional patches without modifing ebuild
 	eapply_user

+	# prebuilt glibc binary files cause python2.7 segfault on musl
+	if use elibc_musl; then
+		rm "${S}"/third_party/python/psutil/psutil/*.so
+		rm "${S}"/third_party/python/psutil/build/lib.linux-x86_64-2.7/psutil/*.so
+		rm "${S}"/third_party/python/psutil/build/temp.linux-x86_64-2.7/psutil/*.o
+	fi
+
 	# Enable gnomebreakpad
 	if use debug ; then
 		sed -i -e "s:GNOME_DISABLE_CRASH_DIALOG=1:GNOME_DISABLE_CRASH_DIALOG=0:g" \

```

**Comment by anarchpenguin** [comment]
I have added an lld and binutils ebuild to my dev overlay, this sets the stacksize to 2MB which means both gcc and clang can build firefox/thunderbird with lto enabled. I have updated 76.0_beta4 ebuild also in my devspace with the changes you have submitted here, so pgo will work as well. I am working with toolchain project to try and get them to add binutils change to tree, if they do not will add to musl overlay. Thanks for the work you have done to help improve the mozilla packages for musl users.
